single-node-refactor/src/Solvers/SGH_solver_3D/src/momentum.cpp

```diff
@@ -93,15 +93,106 @@ void SGH3D::update_velocity(double rk_alpha,
+        } // end for
+
+        // --- calculate the velocity gradient terms ---
+        double inverse_vol = 1.0 / elem_vol(elem_gid);
+
+        // x-dir
+        vel_grad(elem_gid, 0, 0) = (u(0) * b_matrix(0, 0) + u(1) * b_matrix(1, 0)
+            + u(2) * b_matrix(2, 0) + u(3) * b_matrix(3, 0)
+            + u(4) * b_matrix(4, 0) + u(5) * b_matrix(5, 0)
+            + u(6) * b_matrix(6, 0) + u(7) * b_matrix(7, 0)) * inverse_vol;
+
+        vel_grad(elem_gid, 0, 1) = (u(0) * b_matrix(0, 1) + u(1) * b_matrix(1, 1)
+            + u(2) * b_matrix(2, 1) + u(3) * b_matrix(3, 1)
+            + u(4) * b_matrix(4, 1) + u(5) * b_matrix(5, 1)
+            + u(6) * b_matrix(6, 1) + u(7) * b_matrix(7, 1)) * inverse_vol;
+
+        vel_grad(elem_gid, 0, 2) = (u(0) * b_matrix(0, 2) + u(1) * b_matrix(1, 2)
+            + u(2) * b_matrix(2, 2) + u(3) * b_matrix(3, 2)
+            + u(4) * b_matrix(4, 2) + u(5) * b_matrix(5, 2)
+            + u(6) * b_matrix(6, 2) + u(7) * b_matrix(7, 2)) * inverse_vol;
+
+        // y-dir
+        vel_grad(elem_gid, 1, 0) = (v(0) * b_matrix(0, 0) + v(1) * b_matrix(1, 0)
+            + v(2) * b_matrix(2, 0) + v(3) * b_matrix(3, 0)
+            + v(4) * b_matrix(4, 0) + v(5) * b_matrix(5, 0)
+            + v(6) * b_matrix(6, 0) + v(7) * b_matrix(7, 0)) * inverse_vol;
+
+        vel_grad(elem_gid, 1, 1) = (v(0) * b_matrix(0, 1) + v(1) * b_matrix(1, 1)
+            + v(2) * b_matrix(2, 1) + v(3) * b_matrix(3, 1)
+            + v(4) * b_matrix(4, 1) + v(5) * b_matrix(5, 1)
+            + v(6) * b_matrix(6, 1) + v(7) * b_matrix(7, 1)) * inverse_vol;
+        vel_grad(elem_gid, 1, 2) = (v(0) * b_matrix(0, 2) + v(1) * b_matrix(1, 2)
+            + v(2) * b_matrix(2, 2) + v(3) * b_matrix(3, 2)
+            + v(4) * b_matrix(4, 2) + v(5) * b_matrix(5, 2)
+            + v(6) * b_matrix(6, 2) + v(7) * b_matrix(7, 2)) * inverse_vol;
+
+        // z-dir
+        vel_grad(elem_gid, 2, 0) = (w(0) * b_matrix(0, 0) + w(1) * b_matrix(1, 0)
+            + w(2) * b_matrix(2, 0) + w(3) * b_matrix(3, 0)
+            + w(4) * b_matrix(4, 0) + w(5) * b_matrix(5, 0)
+            + w(6) * b_matrix(6, 0) + w(7) * b_matrix(7, 0)) * inverse_vol;
+
+        vel_grad(elem_gid, 2, 1) = (w(0) * b_matrix(0, 1) + w(1) * b_matrix(1, 1)
+            + w(2) * b_matrix(2, 1) + w(3) * b_matrix(3, 1)
+            + w(4) * b_matrix(4, 1) + w(5) * b_matrix(5, 1)
+            + w(6) * b_matrix(6, 1) + w(7) * b_matrix(7, 1)) * inverse_vol;
+
+        vel_grad(elem_gid, 2, 2) = (w(0) * b_matrix(0, 2) + w(1) * b_matrix(1, 2)
+            + w(2) * b_matrix(2, 2) + w(3) * b_matrix(3, 2)
+            + w(4) * b_matrix(4, 2) + w(5) * b_matrix(5, 2)
+            + w(6) * b_matrix(6, 2) + w(7) * b_matrix(7, 2)) * inverse_vol;
+    });  // end parallel for over elem_gid
+
+    return;
+} // end subroutine
+/*
 KOKKOS_FUNCTION
-void SGH3D::get_velgrad(ViewCArrayKokkos<double>& vel_grad,
+void SGH3D::get_velgrad(DCArrayKokkos<double>& vel_grad,
```

Comment:
I think for the PR we should go ahead remove the unused version.  Will help with future merging as well.  Git may not recognize them as separate and cause all sorts of issues.